public/js/src/module/store.js

```diff
@@ -80,8 +80,16 @@ function init({ failSilently } = {}) {
                             keyPath: 'instanceId',
                         },
                         indexes: {
-                            // useful to check if name exists
-                            name: {
+                            // https://github.com/enketo/enketo-express/issues/416
+                            // Version 3 of this database schema had a 'name' index that was set to be unique.
+                            // Although this index was removed in version 4, it will not be automatically deleted in
+                            // existing databases. This is a limitation of db.js. It should not cause significant issues
+                            // as the user will be prompted to edit the record name. Eventually, due to device upgrades,
+                            // or manual indexedDb deletion, this index will probably cease to exist in the world.
+                            //
+                            // to prevent showing the same record name in the queue
+                            recordName: {
+                                keyPath: ['name', 'enketoId'],
```

Comment:
I tried many things to get the old 'name' index removed from an existing database, but I have concluded that it will require either an update to db.js (which will not happen) or a switch to a different indexedDb library.

Reply:
Currently on mobile but thought I'd take a quick glance. I wonder if, instead of removing the `name` index it might be possible/sufficient to still include it but remove its uniqueness constraint?

Reply:
Yes, good idea. Unfortunately, that has the same problem. It won't actually update for existing databases, so it seemed the equivalent to removing it (since we don't actually use the index to search for records anyway).